dali/pipeline/operators/util/pad.cu

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <map>
+#include <vector>
+#include "dali/pipeline/operators/util/pad.h"
+#include "dali/core/static_switch.h"
+#include "dali/pipeline/data/views.h"
+
+namespace dali {
+
+DALI_SCHEMA(Pad)
+    .DocStr(R"code(Pads all samples with `padding_value` in a given `axis`,
+to match the size of the biggest dimension on this axis in the batch.
+The element padding axis is specified with the argument `axis`.
+Supported types: int, float.
+Examples:
+- Batch of 3 1-d samples, padding_value=-1, axis=0
+  input  = [{3, 4, 2, 5, 4},
+            {2, 2},
+            {3, 199, 5}};
+  output = [{3, 4, 2, 5, 4},
+            {2, 2, -1, -1, -1},
+            {3, 199, 5, -1, -1}]
+- Batch of 2 2-d samples, padding_value=42, axis=1
+  input  = [{{1, 2 , 3, 4},
+             {5, 6, 7, 8}},
+            {{1, 2},
+             {4, 5}}]
+  output = [{{1,  2,  3,  4},
+             {5,  6,  7,  8}},
+            {{1,  2, 42, 42},
+             {4,  5, 42, 42}}]
+)code")
+    .NumInput(1)
+    .NumOutput(1)
+    .AddArg("padding_value",
+        R"code(The value to pad the batch with)code",
+        DALI_FLOAT)
+    .AddOptionalArg("axis",
```

Comment:
Rename to `dims` and make it a `vector<int>`. This should default to empty, in which case everything is padded.